File: apps/api/src/pinky_api/models/work_item.py
import uuid
from datetime import datetime

from sqlalchemy import Float, ForeignKey, Index, String
from sqlalchemy.dialects.postgresql import JSONB, UUID
from sqlalchemy.orm import Mapped, mapped_column

from pinky_api.models.base import Base, TimestampMixin, gen_uuid


class WorkItem(Base, TimestampMixin):
    __tablename__ = "work_items"
    __table_args__ = (
        Index("idx_work_items_cluster", "cluster_id", "status"),
        Index("idx_work_items_owner", "owner_id", "status"),
    )

    id: Mapped[uuid.UUID] = mapped_column(UUID, primary_key=True, default=gen_uuid)
    issue_id: Mapped[uuid.UUID | None] = mapped_column(UUID, ForeignKey("issues.id"))
    cluster_id: Mapped[uuid.UUID] = mapped_column(UUID, ForeignKey("cluster_registry.id"), nullable=False)
    title: Mapped[str] = mapped_column(String, nullable=False)
    why_now: Mapped[str | None] = mapped_column(String)
    recommended_next_step: Mapped[str | None] = mapped_column(String)
    status: Mapped[str] = mapped_column(String, server_default="ready")
    owner_id: Mapped[uuid.UUID | None] = mapped_column(UUID, ForeignKey("principals.id"))
    confidence: Mapped[float | None] = mapped_column(Float)
    priority: Mapped[str] = mapped_column(String, server_default="medium")
    labels: Mapped[dict] = mapped_column(JSONB, server_default="{}")
    annotations: Mapped[dict] = mapped_column(JSONB, server_default="{}")
    runbook_url: Mapped[str | None] = mapped_column(String)
    artifact_refs: Mapped[dict] = mapped_column(JSONB, server_default="{}")
    blocked_reason: Mapped[str | None] = mapped_column(String)

File: packages/contracts/src/work-item.ts
export type WorkItemStatus =
  | "ready"
  | "accepted"
  | "in_progress"
  | "blocked"
  | "waiting_for_approval"
  | "done";

export interface WorkItem {
  id: string;
  issue_id: string | null;
  cluster_id: string;
  title: string;
  why_now: string | null;
  recommended_next_step: string | null;
  status: WorkItemStatus;
  owner_id: string | null;
  confidence: number | null;
  priority: "critical" | "high" | "medium" | "low";
  labels: Record<string, string>;
  annotations: Record<string, string>;
  runbook_url: string | null;
  artifact_refs: Record<string, unknown>;
  blocked_reason: string | null;
  created_at: string;
  updated_at: string;
}
